test/sqllogictest/transform/join_index.slt

```diff
@@ -268,34 +268,34 @@ CREATE INDEX bar_idx3 on bar(a + 4);
 query T multiline
 EXPLAIN PLAN FOR
 select foo.b, bar.b, baz.b
-FROM foo, bar, baz
+FROM bar, foo, baz
```

Comment:
In this query, if "foo" came before "bar", then we would not be able to use the index `bar(a+4)`.
The set of equivalence classes `(= #0 #2) (= #4 (#2 + 4))` would get rewritten to `(= #0 #2) (= #4 (#0 + 4))`. Then, JoinImplementation will look for an index on `foo(a+4)` but not an index on `bar(a+4)`.

I have written up in #8002 that JoinImplementation could be changed to identify arrangements on expressions that are equivalent to expressions in `equivalences`.